### ISSUE TITLE
examples\controlnet add only_mid_control=true and sd_locked=false

### ISSUE THREAD

**Issue by geroldmeisinger** [huggingface/diffusers#5070]
**Is your feature request related to a problem? Please describe.**
The [original controlnet training tutorial](https://github.com/lllyasviel/ControlNet/blob/main/docs/train.md#other-options) mentions the options
* `only_mid_control=True`: _This can be helpful when your computation power is limited and want to speed up the training, or when you want to facilitate the "global" context learning._
* `sd_locked=False`: _This will unlock some layers in SD and you will train them as a whole._

**Describe the solution you'd like**
Please make the option available as well.

**Describe alternatives you've considered**
Going back to the original training script without any optimizations implemented.

**Comment by patrickvonplaten** [comment]
cc @williamberman @patil-suraj

**Comment by williamberman** [comment]
Yeah these are neat ideas! I think these would be good options for community PRs, happy to open up to the community :)

**Comment by kelisiya** [comment]
Have any updates about this project ?

**Comment by jiangyuhangcn** [comment]
How to Implement SD_ Locked=False in Diffusers？

**Comment by ValMystletainn** [comment]
I would like to try this issue.
Seemingly, It can be done by set different gradient recording switch and keep it compatible with accelerate.
@williamberman @patrickvonplaten @patil-suraj

**Comment by jon-chuang** [comment]
Seems pretty straightforward, just add an option to freeze/unfreeze certain weights.

**Comment by UmerHA** [comment]
Hi @ValMystletainn, are you still on it? :) Otherwise I'd tackle it.

**Comment by Night1099** [comment]
This is no longer an option in current controlnet train script
